### DISCUSSION TITLE
Does MSAL library prefetch the pages in chrome custom tabs?

### DISCUSSION THREAD

**Discussion by ghost** [AzureAD/microsoft-authentication-library-for-android#1820] [Q&A]
In Chrome custom tabs, the recommendation is to use `warmup()` and `mayLaunchUrl()` to speed up the browser start-up so that the page loads fast.

Ref: https://developer.chrome.com/docs/android/custom-tabs/guide-warmup-prefetch/

I see that the MSAL library uses `warmup()`. Is there a way to hook `mayLaunchUrl()` as well? i.e., let the client app provide/configures the URL and the custom tab implementation in the MSAL library uses it to pre-fetch the pages.

https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/af128744ec6beaf134ecde0cf619dd0d589382d8/msal/src/main/java/com/microsoft/identity/client/MsalChromeCustomTabManager.java#L184

**Comment by negoe** [comment]
@slakshmanan-isn Unfortunately, the library code does not support mayLaunchUrl() as of today.
